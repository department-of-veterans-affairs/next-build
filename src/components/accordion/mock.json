{
  "type": "paragraph--basic_accordion",
  "id": "ca7c4618-6701-4f84-a636-8209c05c9eca",
  "drupal_internal__id": 38313,
  "drupal_internal__revision_id": 284296,
  "langcode": "en",
  "status": true,
  "created": "2021-04-28T01:16:36+00:00",
  "parent_id": "16142",
  "parent_type": "node",
  "parent_field_name": "field_content_block",
  "behavior_settings": [],
  "default_langcode": true,
  "revision_translation_affected": true,
  "content_translation_source": "und",
  "content_translation_outdated": false,
  "content_translation_changed": null,
  "field_header": "What are Vet Centers?",
  "field_rich_wysiwyg": {
    "value": "<p>Vet Centers are small, non-medical, counseling centers conveniently located in your community. They're staffed by highly trained counselors and team members dedicated to seeing you through the challenges that come with managing life during and after the military.</p>\r\n\r\n<p>Whether you come in for one-on-one counseling or to participate in a group session, at Vet Centers you can form social connections, try new things, and build a support system with people who understand you and want to help you succeed.</p>\r\n",
    "format": "rich_text_limited",
    "processed": "<p>Vet Centers are small, non-medical, counseling centers conveniently located in your community. They're staffed by highly trained counselors and team members dedicated to seeing you through the challenges that come with managing life during and after the military.</p>\n<p>Whether you come in for one-on-one counseling or to participate in a group session, at Vet Centers you can form social connections, try new things, and build a support system with people who understand you and want to help you succeed.</p>\n"
  },
  "links": {
    "self": {
      "href": "https://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/jsonapi/paragraph/basic_accordion/ca7c4618-6701-4f84-a636-8209c05c9eca?resourceVersion=id%3A284296"
    }
  },
  "paragraph_type": {
    "type": "paragraphs_type--paragraphs_type",
    "id": "5f3ba120-da59-4c9a-b7d8-5fd72bf3eecd",
    "resourceIdObjMeta": {
      "drupal_internal__target_id": "basic_accordion"
    }
  },
  "relationshipNames": ["paragraph_type"]
}
