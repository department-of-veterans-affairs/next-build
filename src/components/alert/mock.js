export const mockResponse = {
  type: 'paragraph--alert',
  id: '019070f3-9e3e-4c69-83e4-c7eace8c414f',
  drupal_internal__id: 6003,
  drupal_internal__revision_id: 27807,
  langcode: 'en',
  status: true,
  created: '2019-09-04T13:45:29+00:00',
  parent_id: '876',
  parent_type: 'node',
  parent_field_name: 'field_content_block',
  behavior_settings: [],
  default_langcode: true,
  revision_translation_affected: true,
  content_translation_source: 'und',
  content_translation_outdated: false,
  content_translation_changed: null,
  field_alert_heading: 'Blue Water Navy Veterans and Agent Orange',
  field_alert_type: 'information',
  links: {
    self: {
      href: 'https://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/jsonapi/paragraph/alert/019070f3-9e3e-4c69-83e4-c7eace8c414f?resourceVersion=id%3A27807',
    },
  },
  paragraph_type: {
    type: 'paragraphs_type--paragraphs_type',
    id: 'daa6bb41-1b12-46f7-a355-96ceb6fa18dc',
    resourceIdObjMeta: {
      drupal_internal__target_id: 'alert',
    },
  },
  field_alert_block_reference: null,
  field_va_paragraphs: [
    {
      type: 'paragraph--wysiwyg',
      id: 'abb45dcb-b33f-4080-b20f-2f89d4281806',
      drupal_internal__id: 6004,
      drupal_internal__revision_id: 27808,
      langcode: 'en',
      status: true,
      created: '2019-09-04T13:45:29+00:00',
      parent_id: '6003',
      parent_type: 'paragraph',
      parent_field_name: 'field_va_paragraphs',
      behavior_settings: [],
      default_langcode: true,
      revision_translation_affected: true,
      content_translation_source: 'und',
      content_translation_outdated: false,
      content_translation_changed: null,
      field_wysiwyg: {
        value:
          '<div class="usa-alert-body" ><p>Blue Water Navy Veterans are now entitled to a presumption of service connection for illnesses related to Agent Orange exposure. This is a result of <a href="https://www.congress.gov/bill/116th-congress/house-bill/299/text" target="_blank" rel="noopener ">Public Law 116-23</a>, the Blue Water Navy Vietnam Veterans Act of 2019. The law was signed on June 25, 2019, and takes effect on January 1, 2020.</p><p>Read below to learn more about eligibility requirements and how to file a claim. If we denied your claim in the past, you can file a new claim based on this change in law.</p></div>',
        format: 'rich_text',
        processed:
          '<p>Blue Water Navy Veterans are now entitled to a presumption of service connection for illnesses related to Agent Orange exposure. This is a result of <a href="https://www.congress.gov/bill/116th-congress/house-bill/299/text" target="_blank">Public Law 116-23</a>, the Blue Water Navy Vietnam Veterans Act of 2019. The law was signed on June 25, 2019, and takes effect on January 1, 2020.</p><p>Read below to learn more about eligibility requirements and how to file a claim. If we denied your claim in the past, you can file a new claim based on this change in law.</p>',
      },
      links: {
        self: {
          href: 'https://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/jsonapi/paragraph/wysiwyg/abb45dcb-b33f-4080-b20f-2f89d4281806?resourceVersion=id%3A27808',
        },
      },
      resourceIdObjMeta: {
        target_revision_id: 27808,
        drupal_internal__target_id: 6004,
      },
      paragraph_type: {
        type: 'paragraphs_type--paragraphs_type',
        id: '885e4b61-cfd2-44b9-94ae-f068ba2b48b6',
        resourceIdObjMeta: {
          drupal_internal__target_id: 'wysiwyg',
        },
      },
      relationshipNames: ['paragraph_type'],
    },
  ],
  relationshipNames: [
    'paragraph_type',
    'field_alert_block_reference',
    'field_va_paragraphs',
  ],
}
