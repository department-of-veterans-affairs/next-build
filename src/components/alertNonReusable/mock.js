export const mockResponse = {
  type: 'paragraph--non_reusable_alert',
  id: '4eff76c1-5e3e-410b-a56a-64b572f98756',
  drupal_internal__id: 13072,
  drupal_internal__revision_id: 145559,
  langcode: 'en',
  status: true,
  created: '2020-10-01T17:08:59+00:00',
  parent_id: '13073',
  parent_type: 'paragraph',
  parent_field_name: 'field_alert_non_reusable_ref',
  behavior_settings: [],
  default_langcode: true,
  revision_translation_affected: null,
  content_translation_source: 'und',
  content_translation_outdated: false,
  content_translation_changed: null,
  field_alert_heading: 'Sample non-reusable alert',
  field_alert_type: 'information',
  links: {
    self: {
      href: 'https://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/jsonapi/paragraph/non_reusable_alert/4eff76c1-5e3e-410b-a56a-64b572f98756?resourceVersion=id%3A145559',
    },
  },
  paragraph_type: {
    type: 'paragraphs_type--paragraphs_type',
    id: '3d633fc0-97d6-4087-b709-edfa2d1e6a1e',
    resourceIdObjMeta: {
      drupal_internal__target_id: 'non_reusable_alert',
    },
  },
  field_va_paragraphs: [
    {
      type: 'paragraph--wysiwyg',
      id: '2fc092f1-969a-4136-9039-4fc50a9f93c4',
      drupal_internal__id: 13071,
      drupal_internal__revision_id: 141493,
      langcode: 'en',
      status: true,
      created: '2020-10-01T17:09:17+00:00',
      parent_id: '13072',
      parent_type: 'paragraph',
      parent_field_name: 'field_va_paragraphs',
      behavior_settings: [],
      default_langcode: true,
      revision_translation_affected: true,
      content_translation_source: 'und',
      content_translation_outdated: false,
      content_translation_changed: null,
      field_wysiwyg: {
        value: '<p>This is where the text goes for the alert</p>\r\n',
        format: 'rich_text',
        processed: '<p>This is where the text goes for the alert</p>',
      },
      links: {
        self: {
          href: 'https://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/jsonapi/paragraph/wysiwyg/2fc092f1-969a-4136-9039-4fc50a9f93c4?resourceVersion=id%3A141493',
        },
      },
      resourceIdObjMeta: {
        target_revision_id: 141493,
        drupal_internal__target_id: 13071,
      },
      paragraph_type: {
        type: 'paragraphs_type--paragraphs_type',
        id: '885e4b61-cfd2-44b9-94ae-f068ba2b48b6',
        resourceIdObjMeta: {
          drupal_internal__target_id: 'wysiwyg',
        },
      },
      relationshipNames: ['paragraph_type'],
    },
  ],
  relationshipNames: ['paragraph_type', 'field_va_paragraphs'],
}
