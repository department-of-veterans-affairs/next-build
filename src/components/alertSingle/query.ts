import { QueryFormatter, QueryParams } from 'next-drupal-query'
import { DrupalJsonApiParams } from 'drupal-jsonapi-params'
import { ParagraphAlertSingle } from '@/types/drupal/paragraph'
import {
  AlertSingle,
  AlertBlock,
  AlertNonReusable,
} from '@/types/formatted/alert'
import { formatParagraph } from '@/lib/drupal/paragraphs'
import { queries } from '@/data/queries'

export const params: QueryParams<null> = () => {
  return new DrupalJsonApiParams().addInclude([
    'field_alert_block_reference',
    'field_alert_block_reference.field_alert_content',
    'field_alert_non_reusable_ref',
    'field_alert_non_reusable_ref.field_va_paragraphs',
  ])
}

export const formatter: QueryFormatter<ParagraphAlertSingle, AlertSingle> = (
  entity: ParagraphAlertSingle
) => {
  return {
    type: entity.type as AlertSingle['type'],
    id: entity.id,
    entityId: entity.drupal_internal__id,
    alertSelection:
      entity.field_alert_selection as AlertSingle['alertSelection'],
    blockReference: queries.formatData(
      'block--alert',
      entity.field_alert_block_reference
    ) as AlertBlock,
    nonReusableRef: formatParagraph(
      entity.field_alert_non_reusable_ref
    ) as AlertNonReusable,
  }
}
