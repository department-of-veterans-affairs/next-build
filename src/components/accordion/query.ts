import { QueryFormatter } from 'next-drupal-query'
import { ParagraphAccordion } from '@/types/drupal/paragraph'
import { AccordionItem } from '@/components/accordion/formatted-type'
import { getHtmlFromField } from '@/lib/utils/getHtmlFromField'

export const formatter: QueryFormatter<ParagraphAccordion, AccordionItem> = (
  entity: ParagraphAccordion
) => {
  return {
    type: entity.type as AccordionItem['type'],
    id: entity.id,
    entityId: entity.drupal_internal__id,
    header: entity.field_header || null,
    html: getHtmlFromField(entity.field_rich_wysiwyg),
  }
}
