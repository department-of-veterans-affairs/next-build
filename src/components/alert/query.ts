import { QueryFormatter } from 'next-drupal-query'
import { ParagraphAlert } from '@/types/drupal/paragraph'
import { Alert, AlertBlock, AlertType } from '@/components/alert/formatted-type'
import { formatParagraph } from '@/lib/drupal/paragraphs'
import { queries } from '@/lib/drupal/queries'

export const formatter: QueryFormatter<ParagraphAlert, Alert> = (
  entity: ParagraphAlert
) => {
  return {
    type: entity.type as Alert['type'],
    id: entity.id ?? null,
    entityId: entity.drupal_internal__id ?? null,
    alertType: entity.field_alert_type as AlertType,
    heading: entity.field_alert_heading,
    blockReference: queries.formatData(
      'block--alert',
      entity.field_alert_block_reference
    ) as AlertBlock,
    paragraphs: entity.field_va_paragraphs.map((paragraph) =>
      formatParagraph(paragraph)
    ),
  }
}
