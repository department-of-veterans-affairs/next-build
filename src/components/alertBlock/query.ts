// Define the query params for fetching block--alert.
import { BlockAlert } from '@/types/drupal/block'
import { QueryFormatter, QueryParams } from 'next-drupal-query'
import { AlertBlock, AlertType } from '@/components/alert/formatted-type'
import { formatParagraph } from '@/lib/drupal/paragraphs'
import { Wysiwyg } from '@/components/wysiwyg/formatted-type'
import { ExpandableText } from '@/components/expandableText/formatted-type'
import { DrupalJsonApiParams } from 'drupal-jsonapi-params'

export const params: QueryParams<null> = () =>
  new DrupalJsonApiParams().addInclude(['field_alert_content'])

export const formatter: QueryFormatter<BlockAlert, AlertBlock> = (
  entity: BlockAlert
) => {
  if (!entity) {
    return null
  }

  return {
    alertType: (entity.field_alert_type === 'information'
      ? 'info'
      : entity.field_alert_type) as AlertType,
    id: entity.id,
    title: entity.field_alert_title,
    content: formatParagraph(entity.field_alert_content) as
      | Wysiwyg
      | ExpandableText,
    // {
    //   header:
    //     entity.field_alert_content.field_text_expander ??
    //     entity.field_alert_heading,
    //   text: entity.field_alert_content?.field_wysiwyg?.processed ?? '',
    // },
  }
}
