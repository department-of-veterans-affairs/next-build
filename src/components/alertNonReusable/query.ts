// Define the query params for fetching block--alert.
import { QueryFormatter } from 'next-drupal-query'
import { AlertType, AlertNonReusable } from '@/components/alert/formatted-type'
import { ParagraphNonReusableAlert } from '@/types/drupal/paragraph'
import { formatParagraph } from '@/lib/drupal/paragraphs'

export const formatter: QueryFormatter<
  ParagraphNonReusableAlert,
  AlertNonReusable
> = (entity: ParagraphNonReusableAlert) => {
  if (!entity) {
    return null
  }

  return {
    type: entity.type as AlertNonReusable['type'],
    id: entity.id,
    entityId: entity.drupal_internal__id,
    alertType: (entity.field_alert_type === 'information'
      ? 'info'
      : entity.field_alert_type) as AlertType,
    heading: entity.field_alert_heading,
    paragraphs: entity.field_va_paragraphs.map((paragraph) => formatParagraph(paragraph)),
  }
}
