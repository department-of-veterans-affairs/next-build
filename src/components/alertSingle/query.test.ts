/**
 * @jest-environment node
 */

import { queries } from '@/lib/drupal/queries'
import { alertReusable, alertNonReusable } from './mock'
import { params } from './query'

describe('DrupalJsonApiParams configuration', () => {
  test('params function sets the correct include fields', () => {
    const paramsInstance = params()
    const queryString = decodeURIComponent(paramsInstance.getQueryString())
    expect(queryString).toMatch(
      /include=field_alert_block_reference,field_alert_block_reference.field_alert_content,field_alert_non_reusable_ref,field_alert_non_reusable_ref.field_va_paragraphs/
    )
  })
})

describe('alert single formatData', () => {
  test('formats reusable alert', () => {
    const formattedData = queries.formatData(
      'paragraph--alert_single',
      alertReusable
    )
    expect(formattedData).toMatchSnapshot()
  })

  test('formats non-reusable alert', () => {
    const formattedData = queries.formatData(
      'paragraph--alert_single',
      alertNonReusable
    )
    expect(formattedData).toMatchSnapshot()
  })
})
