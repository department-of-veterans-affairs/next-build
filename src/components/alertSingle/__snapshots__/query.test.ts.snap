// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`alert single formatData outputs formatted data 1`] = `
{
  "alertSelection": "R",
  "blockReference": {
    "alertType": "info",
    "content": {
      "html": "<p>Get the latest information about in-person services, claim exams, extensions, paperwork, decision reviews and appeals, and how best to contact us during this time.</p>

<p><a href="/coronavirus-veteran-frequently-asked-questions/#claims-and-applications">Go to our coronavirus FAQs</a></p>",
      "id": "65fda9e9-1bbb-45f2-bc39-d3c659ba30d1",
      "type": "paragraph--wysiwyg",
    },
    "id": "ba9dd3f4-4260-4e21-bbf9-a5cf0db49fc1",
    "title": "You can still file a claim and apply for benefits during the coronavirus pandemic",
  },
  "entityId": 13070,
  "id": "0e71c5cd-a4e2-4dec-9b1e-0ef1c5666f65",
  "nonReusableRef": null,
  "type": "paragraph--alert_single",
}
`;
